src/base/bittorrent/sessionimpl.cpp

```diff
@@ -2824,10 +2824,13 @@ bool SessionImpl::addTorrent_impl(const TorrentDescriptor &source, const AddTorr
         p.trackers.reserve(p.trackers.size() + static_cast<std::size_t>(m_additionalTrackerList.size()));
         p.tracker_tiers.reserve(p.trackers.size() + static_cast<std::size_t>(m_additionalTrackerList.size()));
         p.tracker_tiers.resize(p.trackers.size(), 0);
+        int trackerTier = (p.tracker_tiers.size() > 0 ? p.tracker_tiers.back() : 0);
         for (const TrackerEntry &trackerEntry : asConst(m_additionalTrackerList))
-        {
+        {   
+            if (trackerTier < std::numeric_limits<decltype(trackerTier)>::max())  // prevent overflow
+                ++trackerTier;
```

Comment:
When `trackerTier = 0` then here `trackerTier` becomes `1`. So the loop should be moved to the end of the `for` body.